external/model-preparation-algorithm/mpa_tasks/extensions/datasets/mpa_cls_dataset.py

```diff
@@ -133,3 +137,105 @@ def class_accuracy(self, results, gt_labels):
             cls_acc = np.sum(cls_pred) / len(cls_pred)
             accracies.append(cls_acc)
         return accracies
+
+
+@DATASETS.register_module()
+class MPAMultilabelClsDataset(MPAClsDataset):
+    def get_indices(self, new_classes):
+        return get_old_new_img_indices(self.labels, new_classes, self.ote_dataset)
+    
+    def load_annotations(self):
+        for dataset_item in self.ote_dataset:
+            label = np.zeros(len(self.labels))
+            roi_label = dataset_item.get_roi_labels(self.labels)
+            for lbl in roi_label:
+                label[self.label_idx[lbl.id]] = 1
+            self.gt_labels.append(label)
+        self.gt_labels = np.array(self.gt_labels)
+
+    def evaluate(self,
+                 results,
+                 metric='mAP',
+                 metric_options=None,
+                 indices=None,
+                 logger=None):
+        """Evaluate the dataset.
+        Args:
+            results (list): Testing results of the dataset.
+            metric (str | list[str]): Metrics to be evaluated.
+                Default value is 'mAP'. Options are 'mAP', 'CP', 'CR', 'CF1',
+                'OP', 'OR' and 'OF1'.
+            metric_options (dict, optional): Options for calculating metrics.
+                Allowed keys are 'k' and 'thr'. Defaults to None
+            logger (logging.Logger | str, optional): Logger used for printing
+                related information during evaluation. Defaults to None.
+        Returns:
+            dict: evaluation results
+        """
+        if metric_options is None or metric_options == {}:
+            metric_options = {'thr': 0.5}
+
+        if isinstance(metric, str):
+            metrics = [metric]
+        else:
+            metrics = metric
+        allowed_metrics = ['accuracy-mlc', 'mAP', 'CP', 'CR', 'CF1', 'OP', 'OR', 'OF1']
+        eval_results = {}
+        results = np.vstack(results)
+        gt_labels = self.get_gt_labels()
+        if indices is not None:
+            gt_labels = gt_labels[indices]
+        num_imgs = len(results)
+        assert len(gt_labels) == num_imgs, 'dataset testing results should '\
+            'be of the same length as gt_labels.'
+
+        invalid_metrics = set(metrics) - set(allowed_metrics)
+        if len(invalid_metrics) != 0:
+            raise ValueError(f'metric {invalid_metrics} is not supported.')
+
+        if 'accuracy-mlc' in metrics:
```

Comment:
The name 'accuracy-mlc' doesn't hit me, is it because it conflicts with 'accuracy'?